### PR TITLE
refactor(user.model.spec): reduce callback hell

### PR DESCRIPTION
**Pull request by deltreey**
by passing functions and reducing function wrapping we can keep complexity low and make
everything cleaner and more readable

this is a small change, just the user model tests.  If you guys like this style I can expand upon it, either making this PR larger or by making another.  Let me know. :+1:

### CODE REVIEW

File: app/templates/server/api/user(auth)/user.model.spec(mongooseModels).js [hunk 1]
```diff
@@ -14,18 +14,12 @@ var genUser = function() {
 };
 
 describe('User Model', function() {
-  before(function() {
-    // Clear users before testing
-    return User.removeAsync();
-  });
+  // Clear users before testing
+  before(User.removeAsync);
```
Comment (kingcody) [inline app/templates/server/api/user(auth)/user.model.spec(mongooseModels).js]: @deltreey my apologies, I should have mentioned that the `User` methods will loose context to their parent object, so they'll need to be bound like: `User.removeAsync.bind(User)`

**Comment by deltreey** [comment]
this PR seems to be broken, I'm testing a bit, seeing if travis and my local box are not agreeing on stuff.  hold off on merging this please :)

File: app/templates/server/api/user(auth)/user.model.spec(mongooseModels).js (hunk 1 above)
Comment (Awk34) [inline app/templates/server/api/user(auth)/user.model.spec(mongooseModels).js]: Babel supports `::User.removeAsync` too, right?

**Comment by Awk34** [comment]
I wouldn't really describe this as "callback hell", and could argue that this PR would add *more* complication

**Comment by deltreey** [comment]
I agree.  Should've abandoned this pull req a while ago

**Pull request closed without merging**